src/devices/bituo_technik.ts

```diff
@@ -1,84 +1,321 @@
 import fz from '../converters/fromZigbee';
 import * as exposes from '../lib/exposes';
+import {electricityMeter} from '../lib/modernExtend';
 import * as reporting from '../lib/reporting';
-import {DefinitionWithExtend} from '../lib/types';
+import {DefinitionWithExtend, Fz, KeyValueAny} from '../lib/types';
+import * as utils from '../lib/utils';
+import {postfixWithEndpointName, precisionRound} from '../lib/utils';
 
 const e = exposes.presets;
+const ea = exposes.access;
 
+const bituo_fz = {
+    electrical_measurement: {
```

Comment:
this is a lot of duplication, can't we add a new parameter the the electricityMeter modern extend for this?

Reply:
As I mentioned in this [discussion](https://github.com/Koenkk/zigbee-herdsman-converters/issues/8395), adjusting the parameters of power in extend modern did not work for me, and if I change the electricityMeter modern extend, maybe 2 will affect others, and creating a converter in our brand's device file is the safest option. We will have more products that need to use this converter in the future, so I think it is worth it.

Reply:
@Koenkk 
Hi, any news on the pull request? Let me know if there's anything else needed. Thanks!

Reply:
> I change the electricityMeter modern extend, it might affect others.

This can be avoided by adding a parameters, this just duplicates too much code.

Reply:
But I don't know how to add a new parameter for modern extend...

Reply:
> > I change the electricityMeter modern extend, it might affect others.
> 
> This can be avoided by adding a parameters, this just duplicates too much code.

I checked the code of modern extend, divisor and mutiplier are just indexes, you still use electrical measurement as your converter, so I still need to rewrite the logic of electrical measurement, otherwise it will be the same as the result of using forced now, which does not work for my device. 

In addition, your modern extend does not support total power, I added this in my converter, so I don't think this is "useless duplication of code"

Reply:
> > I change the electricityMeter modern extend, it might affect others.
> 
> This can be avoided by adding a parameters, this just duplicates too much code.

If you insist on asking me to change the modern extend instead of rewriting a "duplicate code" converter, please give me some help, because this problem is very tricky for this device, thank you very much

Reply:
> which does not work for my device.

could you explain what exactly does not work?

> In addition, your modern extend does not support total power, 

I think that can be added to the original modernExtend?

Reply:
> > which does not work for my device.
> 
> could you explain what exactly does not work?
> 
extend: [
electricityMeter({
fzMetering: fz.metering,
fzElectricalMeasurement: bituo_fz.electrical_measurement,
acFrequency: true,
powerFactor: true,
configureReporting: false,
producedEnergy: true,
power: {divisor: 1, multiplier: 1},
}),
], 
the power: {divisor: 1, multiplier: 1} in it does not work for my device, it cannot change my divisor and multiplier

Reply:
I see, in that case we should debug why this doesn't work (as it should). Could you try to add a `console.log()` above https://github.com/Koenkk/zigbee-herdsman-converters/blob/ead17e18f26e824305face85c8b5338de11a7937/src/lib/modernExtend.ts#L1915

and see if that is logged?

Reply:
> I see, in that case we should debug why this doesn't work (as it should). Could you try to add a `console.log()` above
> 
> https://github.com/Koenkk/zigbee-herdsman-converters/blob/ead17e18f26e824305face85c8b5338de11a7937/src/lib/modernExtend.ts#L1915
> 
> and see if that is logged?

I think the cause of this problem has been found in this [discussion](https://github.com/Koenkk/zigbee-herdsman-converters/issues/8395)

Also, could you please tell me how to modify my own modern extend files for zigbee2mqtt? I installed HomeAssistant on linux, and I have never tried to modify these files before. Thanks a lot.

Reply:
> I think what we can do is reset the divisor/multiplier immediately after receiving the attribute report through an `onEvent` handler, see

Thank you very much! I will try it.